### PR TITLE
User analytics

### LINKED ISSUE #19: Perf Review Ideas
for performance reviews, I think it would be useful to see, per person: 

1. how many issues they opened
1.1 in how many issues opened by non-parity members they participated.
1.2 they closed how many issues opened by non-parity members
2. how many PRs they merged
3. how many reviews they did
3.1 how many comments they left per review on avg
4. how many SE questions they answered
5. how many polkadot forum conversations they had.

> issues opened by non-parity members are basically a proxy for external customr support and bugs.

### PR DESCRIPTION
**Pull request by Bullrich**
- Updated project to use yarn 4
- Added ability to fetch information about a specific user and get their metrics
  - Resolves #19
  - Created new workflow for this specific case
- Reduced amount of information obtained from the endpoint to half
  - This is for the [secondary rate limits](https://docs.github.com/en/graphql/overview/rate-limits-and-node-limits-for-the-graphql-api#secondary-rate-limits). It seems that if we obtain 100 values with 100 values each, it counts as 1.000 points, and the limit is 2.000 points, so by halving it we reduce the risk of hitting the limit.

# To do
- [x] Improve the documentation

### CODE REVIEW

File: src/report/user.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  }
+
+  public generateMetrics(
+    prList: PullRequestNode[],
+    issues: IssueNode[],
+  ): UserMetrics {
+    const authorship = this.generateMetricsForPullRequestsAuthorship(prList);
+    const reviews = this.generateMetricsForPullRequestsReviews(prList);
+    const issueMetrics = this.generateIssuesMetrics(issues);
+    return { ...authorship, ...reviews, ...issueMetrics };
```
Comment (mutantcornholio) [inline src/report/user.ts:44]: Are we just joining all issues, reviews and PRs?

Comment (Bullrich) [reply src/report/user.ts:44]: Yes, as the final object is 
```typescript
export interface UserMetrics {
  created: MonthWithMatch[];
  closed: MonthWithMatch[];
  reviewsPerMonth: MonthWithMatch[];
  commentsPerReview: MonthWithMatch[];
  participatedIssuesPerMonth: MonthWithMatch[];
  nonOrgParticipatedIssuesPerMonth: MonthWithMatch[];
}
``` 
and each method return two values of the object

File: .github/workflows/generate-user-metrics.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        type: string
+        required: false
+        default: "polkadot-fellows"
+      repo:
+        description: "Repository"
+        type: string
+        required: false
+        default: "runtimes"
+      user:
+        description: "Repository"
```
Comment (rzadp) [inline .github/workflows/generate-user-metrics.yml]:
```suggestion
        description: "User"
```

**Review by rzadp — COMMENTED** [review]
Would it be possible to gather data for all repos in the organization?

For example if we want to get data for members of our team, there isn't a single repository we can point to, it would make sense to point to the whole of `paritytech`.

File: src/report/user.ts (hunk 1 above)
Comment (mutantcornholio) [reply src/report/user.ts:44]: Ah, okay, I see

**Review by mutantcornholio — APPROVED** [review]

**Comment by Bullrich** [comment]
> Would it be possible to gather data for all repos in the organization?
> 
> For example if we want to get data for members of our team, there isn't a single repository we can point to, it would make sense to point to the whole of `paritytech`.

I wouldn't do it for the whole org, as it can be a **lot** of repositories.

I did #21 so the user can list the desired repos and gather the information about them. That will also work cross org.

**Pull request merged**